todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java

```diff
@@ -80,20 +82,19 @@ protected void onCreate(Bundle savedInstanceState) {
             shouldLoadDataFromRepo = savedInstanceState.getBoolean(SHOULD_LOAD_DATA_FROM_REPO_KEY);
         }
 
-        // Create the presenter
-        mAddEditTaskPresenter = new AddEditTaskPresenter(
-                taskId,
+        mViewModel = new AddEditTaskViewModel(
+                getApplicationContext(),
```

Comment:
oh so above i thought this is created w/ activity context, turns out it is using application context. This is too confusing imo. It should just never receive context. And also, we should retain this ViewModel on configuration change via non-config state.